File: production/client/src/main/scala/kpn/client/components/network/nodes/NetworkNodeFilterCriteria.scala
// Migrated to Angular: network-node-filter-criteria.ts
package kpn.client.components.network.nodes

import kpn.client.filter.TimeFilterKind

case class NetworkNodeFilterCriteria(
  tagged: Option[Boolean] = None,
  definedInNetworkRelation: Option[Boolean] = None,
  definedInRouteRelation: Option[Boolean] = None,
  referencedInRoute: Option[Boolean] = None,
  connection: Option[Boolean] = None,
  roleConnection: Option[Boolean] = None,
  integrityCheck: Option[Boolean] = None,
  integrityCheckFailed: Option[Boolean] = None,
  lastUpdated: TimeFilterKind.Value = TimeFilterKind.ALL
)

File: client/src/app/analysis/network/nodes/network-node-filter-criteria.ts
import { SurveyDateFilterKind } from '@app/kpn/filter';
import { TimestampFilterKind } from '@app/kpn/filter';

export class NetworkNodeFilterCriteria {
  constructor(
    readonly definedInNetworkRelation: boolean = null,
    readonly referencedInRoute: boolean = null,
    readonly connection: boolean = null,
    readonly roleConnection: boolean = null,
    readonly integrityCheck: boolean = null,
    readonly integrityCheckFailed: boolean = null,
    readonly lastUpdated: TimestampFilterKind = TimestampFilterKind.all,
    readonly lastSurvey: SurveyDateFilterKind = SurveyDateFilterKind.all,
    readonly proposed: boolean = null
  ) {}
}
